src/main/java/com/battleships/battleships/BattleshipGridConfig.java

```diff
@@ -3,7 +3,7 @@
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-
+//
```

Comment:
Ahh Spring ;-)
I appreciate it you've used it as a learning opportunity but it all feels very heavy-wait for this application.
Anyway rather than declaring every component as a Spring Bean you could perhaps have simply new-ed some of the dependencies up within a single bean declaration.